### PR TITLE
[ci.yaml] Add platform_args, properties, recipe, and timeout information

### PR DESCRIPTION
**Pull request by CaseyHillers**
This makes it possible to generated the luci config builders from this file.

https://github.com/flutter/flutter/issues/84998

Related infra CL: https://flutter-review.googlesource.com/c/infra/+/15360

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I added new tests to check the change I am making or feature I am adding, or Hixie said the PR is test-exempt. See [testing the engine] for instructions on
writing and running engine tests.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I signed the [CLA].
- [x] All existing and new tests are passing.

### CODE REVIEW

**Review by keyonghan — APPROVED** [review]
LGTM

File: .ci.yaml [hunk 1]
```diff
@@ -8,46 +8,113 @@
...
   - master
 
+platform_properties:
+  linux:
+    properties:
+      caches: builder_linux_engine:builder
+      os: Linux
+  mac:
+    properties:
+      caches: flutter_cocoapods:cocoapods, osx_sdk:old_osx_sdk, builder_mac_engine:builder
```
Comment (CaseyHillers) [inline .ci.yaml:18]: @christopherfujino does this format seem ok to you? In the flutter/infra starlark code, I special case "caches" and split it by `(", ")` and `(":")` to get the `name`, `path` for `swarming.cache`.

Comment (christopherfujino) [reply .ci.yaml:18]: - `flutter_cocoapods:cocoapods, osx_sdk:old_osx_sdk, builder_mac_engine:builder` is a single string, correct? if so, can you make that explicit with double quotes?

I was going to suggest making this stringified list parseable JSON, like:

```
caches: "[\"flutter_cocoapods:cocoapods\", \"osx_sdk:old_osx_sdk\"]"
```

But then I realize that there is no JSON.parse library in lucicfg, so we'd have to hand parse it anyway.

Comment (christopherfujino) [reply .ci.yaml:18]: Oh, maybe it is implemented now? https://chromium.googlesource.com/infra/luci/luci-go/+/refs/heads/main/lucicfg/starlark/stdlib/json_doc.star#80

I tried using this locally and it works...

```
config_str = """
{
  "platform_properties": {
    "linux": {
      "properties": {
        "caches": "builder_linux_engine:builder",
        "os": "Linux"
      }
    },
    "mac": {
      "properties": {
        "caches": "flutter_cocoapods:cocoapods, osx_sdk:old_osx_sdk, builder_mac_engine:builder"
      }
    }
  }
}
"""
config_dic = json.decode(config_str)
caches = config_dic["platform_properties"]["mac"]["properties"]["caches"]
print(caches.split(", ")[1].split(":")) # ["osx_sdk", "old_osx_sdk"]
```

File: .ci.yaml [hunk 2]
```diff
@@ -8,46 +8,124 @@
...
+    timeout: 60
     scheduler: luci
 
   - name: Linux Android Debug Engine
     builder: Linux Android Debug Engine
+    recipe: engine
+    properties:
+      add_recipes_cq: "true"
+      build_android_debug: "true"
+      android_sdk_license: \n24333f8a63b6825ea9c5514f83c2829b004d1fee
```
Comment (christopherfujino) [inline .ci.yaml:55]: what's with the leading newline?

Comment (CaseyHillers) [reply .ci.yaml:55]: Based on https://stackoverflow.com/a/38381577, it seems standard for the android sdk license to prefix with newline (this is just a copy and paste of the existing property)

Comment (christopherfujino) [reply .ci.yaml:55]: interesting

**Pull request merged**